src/fixLockfileIntegrity.ts

```diff
@@ -26,37 +26,57 @@ const prettierInitialConfig: prettier.Options = {
 
 type NPMJS_METADATA_PARTIAL = { dist: { integrity: string } };
 
+export const formHashApiUrl = (registry: URL, packageName: string, packageVersion: string) => {
+    const integrityUrlPath: string = [packageName, packageVersion].join("/");
+    return new URL(integrityUrlPath, registry);
+};
+
 /**
  * Fetches for sha512 integrity for a package version
  *
  * @param registry          registry
  * @param packageName       package name
  * @param packageVersion    package version
  */
-const getIntegrity = async (registry: URL, packageName: string, packageVersion: string): Promise<string> => {
-    const integrityUrlPath: string = [...registry.pathname.split("/").filter(Boolean), packageName, packageVersion].join("/");
-    const url: URL = new URL(integrityUrlPath, registry);
+export const getIntegrity = async (registry: URL, packageName: string, packageVersion: string): Promise<string> => {
+    const url: URL = formHashApiUrl(registry, packageName, packageVersion);
     const options: OptionsOfJSONResponseBody = {
         responseType: "json",
         throwHttpErrors: false,
         headers: {
             Accept: "application/json"
         }
     };
-    const metadata: GotResponse<NPMJS_METADATA_PARTIAL> = await got.get<NPMJS_METADATA_PARTIAL>(url, options);
+    let metadata: Response<NPMJS_METADATA_PARTIAL>;
+    try {
+        metadata = await got.get<NPMJS_METADATA_PARTIAL>(url, options);
+    }
+    catch (error) {
+        logger.warn(`${chalk.red("Error retrieving response from API:")} ${chalk.blue(url.toString())} - ${chalk.red(error.message)}`);
+        return undefined;
+    }

```

Comment:
I saw you added a test that mocks rejection by `got`, but did you see this in a real flow? 
Can you simulate an example where `throwHttpErrors: false` is not enough and `got` still throws? (no need to add a test, an example URL is enough, I'll test locally)

Reply:
The error I was seeing was `Error: getaddrinfo ENOTFOUND` and the whole script would crash. So for any domain that doesn't resolve.

Reply:
Got it!
(pun intended)

Reply:
Let me double check this on my end

Reply:
Can confirm it catches the error as expected. However I noticed another unrelated bug - I'll create a separate issue for that :)

Reply:
Okay created an issue for that https://github.com/yoavain/fix-lockfile-integrity/issues/235 - I think we don't have to worry about that scenario very much as it is rare.